shared/request_caching.go

```diff
@@ -36,9 +36,14 @@ func (w *cachingResponseWriter) Write(data []byte) (int, error) {
 		w.statusCode = http.StatusOK
 	}
 
-	_, w.err = w.b.Write(data)
+	n, err := w.delegate.Write(data)
+	if err != nil {
+		return n, err
+	}
+
+	_, w.bufErr = w.b.Write(data)
```

Comment:
If you simply `return w.b.Write(data)` here, you won't need `bufErr` at all, right?

Or do you think error writing to buffer needs to be handled differently (from error writing to response)?

Reply:
We want the consistency from decorated Write() fails iff delegate write failed. Problems with the decorator are handled externally.